### DISCUSSION TITLE
How to congiure or enable the TLS1.3 with specific cipher .

### DISCUSSION THREAD

**Discussion by rajagopala333** [babelouest/ulfius#266] [Q&A]
struct MHD_Daemon *daemon = NULL;
                                        int retvalue = 5;
                                        char one_char ;
                                        char *two_char = &one_char;
                                        daemon =(agent->restInstance).mhd_daemon;
                                        /*retvalue=gnutls_priority_init (&daemon->priority_cache,"NORMAL",NULL);*/
                                        retvalue=gnutls_priority_init (&daemon->priority_cache,"NORMAL:+VERS-TLS-ALL",&two_char);

and connecting from openssl client with the below command\
**openssl s_client -connect localhost:8082  -tls1_3**
CONNECTED(00000003)
4077521B5D7F0000:error:0A000126:SSL routines:(unknown function):unexpected eof while reading:ssl/record/rec_layer_s3.c:304:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 232 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
This TLS version forbids renegotiation.
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---

**Comment by babelouest** [comment]
This doesn't seem related to ulfius, I'm closing this discussion then.

**Comment by rajagopala333** [comment]
ok thank you, How to configure the TLSversions and ciphers through ulfius
